pkg/operator/controller/callbacks.go

```diff
@@ -355,3 +360,47 @@ func restoreOlderVersions(currentCrd, desiredCrd *extv1.CustomResourceDefinition
 	}
 	return desiredCrd
 }
+
+func reconcilePvcMutatingWebhook(args *callbacks.ReconcileCallbackArgs) error {
+	if args.State != callbacks.ReconcileStatePostRead {
+		return nil
+	}
+
+	crd := args.CurrentObject.(*extv1.CustomResourceDefinition)
```

Comment:
I think it would make more sense to link this with the "cdi-apiserver" deployment.  If for some reason cdi-apiserver is not created we would not want to create a webhook

Reply:
But then how would you delete the webhook if "cdi-apiserver" somehow disappears? not sure these cases are relevant as you cannot do anything without "cdi-apiserver" (can you?), and we reconcile for its eventual consistency.

Reply:
Yeah interesting point, it is doubtful that we will ever not deploy cdi-apiserver but I still think it is better to time the creation of the webhook close to the creation of cdi-apiserver ideally right before it

Reply:
fixed